xinference/api/oauth2/auth_service.py

```diff
@@ -40,13 +41,33 @@ def __init__(self, auth_config_file: Optional[str]):
     def config(self):
         return self._config
 
+    @staticmethod
+    def is_legal_api_key(key: str):
+        pattern = re.compile("^[sk]{2}-[a-zA-Z0-9]{48}$")
+        if re.match(pattern, key):
+            return True
+        else:
+            return False
+
     def init_auth_config(self):
         if self._auth_config_file:
             config: AuthStartupConfig = parse_file_as(
                 path=self._auth_config_file, type_=AuthStartupConfig
             )
+            total_keys = set()
             for user in config.user_config:
                 user.password = get_password_hash(user.password)
+                if len(set(user.api_keys)) != len(user.api_keys):
+                    raise ValueError("User has duplicate Api-Keys")
```

Comment:
Does this logic be duplicated with the code below.
The below code can detect the situation that having duplicate api-keys of a user.